src/lib/dump_dir.c

```diff
@@ -755,8 +758,8 @@ uid_t dd_get_owner(struct dump_dir *dd)
 
     if (ret < 0)
     {
-        if (ret != -ENOENT)
-            return ret;
+        if (errorno && ret != -ENOENT)
+            *errorno = ret;
```

Comment:
Thinking out loud: This is a bit different behaviour, isn't it? The previous code bailed immediately, while now only the error code is set but execution continues. It is terminated soon anyway, but the log message might be confusing.

Does it affect anything down the lane?